src/provider/provider_fixed_memory.c

```diff
@@ -0,0 +1,335 @@
+                                              void *highPtr, size_t totalSize) {
+    (void)provider;
+    (void)lowPtr;
+    (void)highPtr;
+    (void)totalSize;
+    return UMF_RESULT_SUCCESS;
+}
+
+static umf_result_t fixed_initialize(void *params, void **provider) {
+    umf_result_t ret;
+
+    if (params == NULL) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    umf_fixed_memory_provider_params_t *in_params =
+        (umf_fixed_memory_provider_params_t *)params;
+
+    if (in_params->ptr == NULL) {
+        LOG_ERR("memory pointer is missing");
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    if (in_params->size == 0) {
+        LOG_ERR("memory size is 0");
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    fixed_memory_provider_t *fixed_provider =
+        umf_ba_global_alloc(sizeof(*fixed_provider));
+    if (!fixed_provider) {
+        return UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY;
+    }
+
+    memset(fixed_provider, 0, sizeof(*fixed_provider));
+
+    coarse_params_t coarse_params = {0};
+    coarse_params.provider = fixed_provider;
+    coarse_params.page_size = utils_get_page_size();
+    // The alloc callback is not available in case of the fixed provider
+    // because it is a fixed-size memory provider
+    // and the entire memory is added as a single block
+    // to the coarse library.
+    coarse_params.cb.alloc = NULL;
+    coarse_params.cb.free = NULL; // not available for the fixed provider
+    coarse_params.cb.split = fixed_allocation_split_cb;
+    coarse_params.cb.merge = fixed_allocation_merge_cb;
+
+    coarse_t *coarse = NULL;
+    ret = coarse_new(&coarse_params, &coarse);
+    if (ret != UMF_RESULT_SUCCESS) {
+        LOG_ERR("coarse_new() failed");
+        goto err_free_fixed_provider;
+    }
+
+    fixed_provider->coarse = coarse;
+
+    fixed_provider->base = in_params->ptr;
+    fixed_provider->size = in_params->size;
+
+    // add the entire memory as a single block
+    ret = coarse_add_memory_fixed(coarse, fixed_provider->base,
+                                  fixed_provider->size);
+    if (ret != UMF_RESULT_SUCCESS) {
+        LOG_ERR("adding memory block failed");
+        goto err_coarse_delete;
+    }
+
+    *provider = fixed_provider;
+
+    return UMF_RESULT_SUCCESS;
+
+err_coarse_delete:
+    coarse_delete(fixed_provider->coarse);
+err_free_fixed_provider:
+    umf_ba_global_free(fixed_provider);
+    return ret;
+}
+
+static void fixed_finalize(void *provider) {
+    fixed_memory_provider_t *fixed_provider = provider;
+    coarse_delete(fixed_provider->coarse);
+    umf_ba_global_free(fixed_provider);
+}
+
+static umf_result_t fixed_alloc(void *provider, size_t size, size_t alignment,
+                                void **resultPtr) {
+    fixed_memory_provider_t *fixed_provider =
+        (fixed_memory_provider_t *)provider;
+    size_t page_size = utils_get_page_size();
+    if (alignment < page_size) {
+        alignment = page_size;
+    }
```

Comment:
It should be removed, because it is done in the coarse library (after having merged https://github.com/oneapi-src/unified-memory-framework/pull/981)